### PR TITLE
Fixed bug with line number inference

### LINKED ISSUE #1773: inference failure with "bring air to end of row 9 and row 15"
`bring air to end of row 9 and row 15` is not equivalent to `bring air to end of row 9 and end of row 15`, but it should be.

Haven't experimented or debugged at all beyond that.

### PR DESCRIPTION
**Pull request by AndreasArvidsson**
Fixes #1773

## Checklist

- [x] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [/] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet)
- [/] I have not broken the cheatsheet

### CODE REVIEW

File: packages/cursorless-engine/src/core/inferFullTargetDescriptor.ts [hunk 1]
```diff
@@ -142,10 +144,15 @@ function getPreservedModifiers(
...
     ) ?? [];
-  if (preservedModifiers.length !== 0) {
-    return preservedModifiers;
-  }
-  // In the absence of any other modifiers line number marks are infer as a containing line scope
+  return preservedModifiers.length !== 0 ? preservedModifiers : undefined;
+}
+
+/**
+ * In the absence of any other modifiers line number marks are infer as a containing line scope
```
Comment (pokey) [inline packages/cursorless-engine/src/core/inferFullTargetDescriptor.ts:151]:
```suggestion
 * `"row five past air"` => `"row five past line air"`
```

File: packages/cursorless-engine/src/core/inferFullTargetDescriptor.ts [hunk 2]
```diff
@@ -110,6 +110,8 @@ function inferPrimitiveTarget(
   const modifiers =
     getPreservedModifiers(target) ??
     getPreviousPreservedModifiers(previousTargets) ??
+    getLineNumberMarkModifiers(target) ??
```
Comment (pokey) [inline packages/cursorless-engine/src/core/inferFullTargetDescriptor.ts:113]: Is this necessary?

**Review by pokey — COMMENTED** [review]
Looks good with minor comments

File: packages/cursorless-engine/src/core/inferFullTargetDescriptor.ts (hunk 2 above)
Comment (AndreasArvidsson) [reply packages/cursorless-engine/src/core/inferFullTargetDescriptor.ts]: Nah

**Review by pokey — APPROVED** [review]
👍

**Pull request merged**